src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -190,90 +193,114 @@ public void AddHighlightMarker(string highlightColor)
 
             // If we've already tagged this document, then we're done.
-            if (this._tag != null)
+            if (this.tag != null)
             {
                 return true;
             }
 
-            // If this document doesn't have anything to do with this result marker, then
-            // skip tagging.
-            if (vsWindowFrame == null ||
-                string.IsNullOrEmpty(documentName) ||
-                string.IsNullOrEmpty(this.FullFilePath) ||
-                string.Compare(documentName, this.FullFilePath, StringComparison.OrdinalIgnoreCase) != 0)
-            {
-                return false;
-            }
-
-            IComponentModel componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
-            if (componentModel == null)
+            if (!this.TryMapRegion())
             {
                 return false;
             }
 
-            if (!SdkUIUtilities.TryGetTextViewFromFrame(vsWindowFrame, out IVsTextView vsTextView))
+            if (!textBuffer.Properties.TryGetProperty(typeof(SarifLocationTagger), out ISarifLocationTagger tagger))
             {
                 return false;
             }
 
-            // Call a bunch of functions to get the WPF text view so we can perform the highlighting only
-            // if we haven't yet
-            if (!SdkUIUtilities.TryGetWpfTextView(vsTextView, out IWpfTextView wpfTextView))
-            {
-                return false;
-            }
-
-            ISarifLocationProviderFactory sarifLocationProviderFactory = componentModel.GetService<ISarifLocationProviderFactory>();
-            _tagger = sarifLocationProviderFactory.GetTextMarkerTagger(wpfTextView.TextBuffer);
-            _tagger.TryGetTag(Region, _runIndex, out ISarifLocationTag existingTag);
+            tagger.TryGetTag(this.mappedRegion, this.runIndex, out ISarifLocationTag existingTag);
 
             if (existingTag == null)
             {
-                if (!TryCreateTextSpanWithinDocumentFromSourceRegion(this.Region, vsWindowFrame, out TextSpan tagSpan))
+                if (!TryCreateTextSpanWithinDocumentFromSourceRegion(this.mappedRegion, textBuffer, out TextSpan tagSpan))
                 {
                     return false;
                 }
 
-                _tag = _tagger.AddTag(Region, tagSpan, _runIndex, new TextMarkerTag(Color));
+                this.tag = tagger.AddTag(this.mappedRegion, tagSpan, this.runIndex, new TextMarkerTag(Color));
             }
             else
             {
-                _tag = existingTag;
+                this.tag = existingTag;
             }
 
 
-            _tag.CaretEnteredTag += CaretEnteredTag;
-            _wpfTextView = wpfTextView;
-            _vsWindowFrame = vsWindowFrame;
-            _wpfTextView.Closed += this.TextViewClosed;
+            if (!SdkUIUtilities.TryGetActiveViewForTextBuffer(this.tag.DocumentPersistentSpan.Span.TextBuffer, out IWpfTextView wpfTextView))
```

Comment:
>(this.tag.DocumentPersistentSpan.Span.TextBuffer [](start = 61, length = 48)

This can just be the passes in text buffer, do not need to use the one from the persistent span